lib/ansible/modules/identity/kinit.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.0',
+    'supported_by': 'community',
+    'status': ['preview']
+}
+
+DOCUMENTATION = '''
+---
+module: kinit
+version_added: "2.4"
+author: "Ali (@bincyber)"
+short_description: Obtain a Kerberos ticket-granting ticket
+description:
```

Comment:
I'm not aware of any security tradeoffs in regards to how the password is supplied to the module. It's exactly the same as how passwords are supplied to other modules that support a password parameter: `no_log` is used to prevent the password from being logged or displayed in verbose mode.

However, I will update the documentation to make a note of why using this module is preferably to expect/shell from a security perspective.